docs/pages/guides/DateAdapterProp.example.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
```

Comment:
What do you think of using `React.useState`?

```suggestion
import React from 'react';
```

Reply:
It is even more invalid, somewhere I thought that this kind of import will be deprecated. We are primarily using `import * as React`, but in examples using this notation. 
I have no preference and can change

Reply:
There is a related thread on this topic at https://github.com/mui-org/material-ui/pull/19802.
What approach have you seen the more frequently used by TypeScript users on one side? And by JavaScript users on the other?

There are no clear guidelines by React on the matter, but I could find these two examples:
- https://reactjs.org/docs/hooks-intro.html 
- https://www.typescriptlang.org/docs/handbook/react-&-webpack.html

Reply:
For typescript codebase, it is much easier to use `import *  as React` because a lot of internal types are also exported, like `React.ReactNode`, `React.HtmlProps` and so on.